src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Aot.targets

```diff
@@ -123,11 +123,25 @@ They run in a context of an inner build with a single $(RuntimeIdentifier).
         LLVMPath="$(_LLVMPath)"
         LdName="$(_LdName)"
         LdFlags="$(_LdFlags)"
+        CollectTrimmingEligibleMethods="$(AndroidStripILAfterAOT)"
+        TrimmingEligibleMethodsOutputDirectory="$(IntermediateOutputPath)tokens"
         WorkingDirectory="$(MSBuildProjectDirectory)"
         AotArguments="$(AndroidAotAdditionalArguments)">
       <Output TaskParameter="CompiledAssemblies" ItemName="_MonoAOTCompiledAssemblies" />
       <Output TaskParameter="FileWrites"         ItemName="FileWrites" />
     </MonoAOTCompiler>
+    <ILStrip
+        Condition=" '$(AndroidStripILAfterAOT)' == 'true' "
+        TrimIndividualMethods="true"
+        Assemblies="@(_MonoAOTCompiledAssemblies)"
+        DisableParallelStripping="$(_DisableParallelAot)">
+      <Output TaskParameter="TrimmedAssemblies" ItemName="_ILStripTrimmedAssemblies" />
+    </ILStrip>
+    <Move
```

Comment:
This `<Move/>` feels absurd to me: `@(_ILStripTrimmedAssemblies)` is an output parameter, so I would normally intuit/assume that `%(_ILStripTrimmedAssemblies.Identity)` is the file that was created.  However, this `<Move/>` implies that `%(_ILStripTrimmedAssemblies.Identity)` has *not* in fact been created, and instead what was created is `%(_ILStripTrimmedAssemblies.TrimmedAssemblyFileName)`, which must be moved to `%(_ILStripTrimmedAssemblies.Identity)`.

Does this make sense to anyone?  Why was `<ILStrip/>` defined this way?

Reply:
The goal of `ILStrip` is to trim IL code away from the assemblies which were AOT'ed. However, not all AOT'ed assemblies might be trimmable. That's why we have `_ILStripTrimmedAssemblies`. `ILStrip` also doesn't overwrite existing assemblies with the trimmed ones. It is designed this way, in case `%(_ILStripTrimmedAssemblies.Identity)` are not the ones that you want to replace with the trimmed ones. They could be the ones which live in other directories. And `ILStrip` could possibly has no knowledge of that.

For `System.Private.CoreLib.dll`, 
`%(_ILStripTrimmedAssemblies.Identity)` is `System.Private.CoreLib.dll`
`%(_ILStripTrimmedAssemblies.TrimmedAssemblyFileName)` is `System.Private.CoreLib_trimmed.dll`

Reply:
@fanyang-mono: how does `<ILStrip/>` know which method bodies it can remove?  I see that the `<MonoAOTCompiler/>` invocation was updated to add a `MonoAOTCompiler.TrimmingEligibleMethodsOutputDirectory` property, but how does `<ILStrip/>` konw about the `$(IntermediateOutputPath)tokens` directory?

Reply:
The output `_MonoAOTCompiledAssemblies` from `MonoAOTCompiler` has a metadata `MethodTokenFile`. It stores the trimmable method token for the corresponding assembly.